plugins/modules/helm.py

```diff
@@ -612,17 +635,39 @@ def main():
         if chart_version is not None:
             helm_cmd += " --version=" + chart_version
 
-        if chart_repo_url is not None:
+        if chart_repo_url:
             helm_cmd += " --repo=" + chart_repo_url
 
         # Fetch chart info to have real version and real name for chart_ref from archive, folder or url
         chart_info = fetch_chart_info(module, helm_cmd, chart_ref)
 
+        if dependency_update:
+            if not chart_info.get('dependencies'):
+                module.warn("There is no dependencies block defined in Chart.yaml. Dependency update will not be performed. "
+                            "Please consider add dependencies block or disable dependency_update to remove this warning.")
+
+            # Can't use '--dependency-update' with 'helm upgrade' that is the
+            # default chart install method, so if chart_repo_url is defined
+            # we can't use the dependency update command. But, in the near future
+            # we can get rid of this method and use only '--dependency-update'
+            # option. Please see https://github.com/helm/helm/pull/8810
+            if not chart_repo_url:
+                run_dep_update(module, helm_cmd_common, chart_ref)
```

Comment:
please consider returning at least the result code ``rc`` which will validate that the dependency update has succeed or return error message if not